teps/0007-conditions-beta.md

```diff
@@ -0,0 +1,612 @@
+-
+
+<!-- ### User Experience (optional) -->
+<!--
+Consideration about the user experience. Depending on the area of change,
+users may be `Task` and pipeline editors, they may trigger `Task` and pipeline
+runs or they may be responsible for monitoring the execution of runs,
+via CLI, dashboard or a monitoring system.
+
+Consider including folks that also work on CLI and dashboard.
+-->
+<!-- - For simple `Conditions`, the user experience would drastically improve because users would only have to pass in the string expression using `CEL` as described above. The user experience would be worse only when they have to build their own `CustomTask` to use a different language, but there are plans to provide frameworks to make that process easier.
+- For complex `Conditions`, the user experience would improve in terms of consistency and learnability because the users would have more experience working with `Tasks`. 
+- For failure handling using a boolean, it would have a default value that represents the current behavior and users who want to continue executing the branch have to provide only one boolean value.  -->
+
+<!-- ### Performance (optional) -->
+<!--
+Consideration about performance.
+What impact does this change have on the start-up time and execution time
+of task and pipeline runs? What impact does it have on the resource footprint
+of Tekton controllers as well as task and pipeline runs?
+
+Consider which use cases are impacted by this change and what are their
+performance requirements.
+-->
+
+<!-- ## Design Details -->
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+## Test Plan
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+- To test correct handling of `Tasks` used as `Conditions`, we'll provide unit tests and e2e tests. Before we migrate to using `Tasks` exclusively, the tests will assert that both are supported. 
+- To test the  `MatchExpressions`, we'll provide unit tests and e2e tests. 
+- To test the CelRun `CustomTasks`, we'll provide unit tests and simple e2e tests that could install a simple CelRun type and controller and assert that a `Task` that references that CelRun type correctly evaluates the string expressions.
```

Comment:
im thinking for the CelRun controller we'd probably make another TEP eventually as well - b/c there is a probably a lot to flesh out around the CEL support specifically, so we probably dont need to go into too much detail here, we could just mention another TEP would be coming